### PR TITLE
I18n: Keeping translations updated

### PR DESCRIPTION
**Pull request by bkeepers**
@webknjaz [commented in #288](https://github.com/github/open-source-guide/issues/288#issuecomment-283927635):

> there's additional concern, which bothers me for a while: fixes or improvements to original articles go to this repo and there's no way to automatically confirm that translated version still corresponds original one.

I met with @alebourne and @zeke last week to talk more about how they plan to handle localization for several other GitHub projects. Long term, the plan is to adopt a translation management tool similar to what was [discussed in #295](https://github.com/github/open-source-guide/pull/295#issuecomment-281414390), which keeps track of which files need updated as content changes. They are well into the process of evaluating options and I'm following their progress closely.

Near term, I'd like to make it as easy as possible for to keep translations up to date, and I think git can help us here. Here's what I propose:

1. Each translation maintainer will occasionally run `$ script/sync-translation` to merge the latest changes from upstream and open a Pull Request on their fork.
2. If files requiring translation have been modified, they will be added to a checklist in the Pull Request with links to the diff.
3. Once all files have been updated, merge the pull request.

This will require the translation authors to manually scan the diffs and make the updates, but it at least provides a lightweight process for doing it so all translations are managed the same.

I would love to hear feedback on this approach. 

cc @nandomoreirame @lijiangsheng1 #295

### CODE REVIEW

**Comment by bkeepers** [comment]
I pushed an initial hacky version of the `script/sync-translation` script. Right now it just echos a bunch of stuff to the command line and doesn't actually do anything, but here's roughly what it will do:

- set up the `upstream` remote if it doesn't already exist and fetch changes
- `git checkout -b sync-$SHA origin/gh-pages`
- `git merge --no-edit upstream/gh-pages`
- Open a PR with a list of files that need translated.

Here is the PR template it will generate:

---

This pull request syncs changes from upstream.

[View changes](https://github.com/github/open-source-guide/compare/0fc50236af4da7a5a339f9681f7c30634f415120...44203979fb97d3ed8ad1bd1b1a59e17357360d31)

Check the changes in each of these files and update the translation accordingly:

- [ ] _articles/en-US/best-practices.md
- [ ] _articles/en-US/building-community.md
- [ ] _articles/en-US/code-of-conduct.md
- [ ] _articles/en-US/finding-users.md
- [ ] _articles/en-US/getting-paid.md
- [ ] _articles/en-US/how-to-contribute.md
- [ ] _articles/en-US/leadership-and-governance.md
- [ ] _articles/en-US/legal.md
- [ ] _articles/en-US/metrics.md
- [ ] _articles/en-US/starting-a-project.md
- [ ] _data/locale/en-US.yml

File: docs/translations.md [hunk 1]
```diff
@@ -14,6 +14,15 @@ If there's not, then today is your day to lead this effort! Here's how to start:
 
 ## Keeping a translation updated
 
-TODO
+Git is pretty good at making it easy to see what files have changed. We'll try to make it as easy as possible for you to keep your translation up to date.
+
+These directions assume you have two [remotes](https://git-scm.com/docs/git-remote) configured for your local clone of is repository.
+
+0. `origin` - the translation fork. If you didn't originally clone this from the fork, you can update it with `git mote set-url origin https://github.com/[yourname]/open-source-guide.git`
```
Comment (webknjaz) [inline docs/translations.md]: **RE**mote

File: script/sync-translation [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+BRANCH="sync-${HEAD:0:8}"
+
+TRANSLATABLE_FILES="
+  _articles/en-US
+  _data/locale/en-US.yml
+"
+
+## Create a new branch and merge in the changes from upstream.
+# TODO: remove echos
+echo git co -b $BRANCH $BASE
```
Comment (webknjaz) [inline script/sync-translation]: I guess, `co` stands for `commit`, right?

File: script/sync-translation [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+# BASE=$(git rev-parse $REMOTE/gh-pages)
+# TODO: replace v with ^
+BASE=0fc50236af4da7a5a339f9681f7c30634f415120 # randomly selected old commit for testing
+HEAD=$(git rev-parse upstream/i18n)
+BRANCH="sync-${HEAD:0:8}"
+
+TRANSLATABLE_FILES="
+  _articles/en-US
+  _data/locale/en-US.yml
+"
```
Comment (webknjaz) [inline script/sync-translation]: What about `_config.yml`? It's got languages list update

**Comment by bkeepers** [comment]
@webknjaz thanks for the review.

The script has been updated and here is an example of it in action: https://github.com/brandon/opensource.guide/pull/3

I would love help testing it. You can run it on your local checkout like this:

```
bash <(curl -sL https://github.com/github/opensource.guide/raw/df24702b82bbabdf27bf4eba8cc2a3171a3ce8a3/script/sync-translation)
```

**Comment by sotayamashita** [comment]
@bkeepers 
I have contributed to OSS including Node.js and I also translated their docs and I have the same problem which @webknjaz commented in #288 so we have created a tool called [GitLocalize](https://gitlocalize.com/). It's like Crowdin but it follows a flow which we always do in GitHub. I am taking about fork and commit/pull request. I think it will help you translate and keep the translated ones updated as well. I can help you try it. If you have any question, Please let me know.

**Comment by bkeepers** [comment]
Thanks @sotayamashita, I passed this on to our team that is evaluating localization tools.

**Comment by sotayamashita** [comment]
@bkeepers I appreciate for replying me. If your team have any question or help, please feel free to ask me via sota@locki.io.

**Pull request merged**